src/DateTimeParser.jl

```diff
@@ -118,32 +113,33 @@ function Base.parse(::Type{Date}, datetimestring::AbstractString; fuzzy::Bool=fa
         return default
     end
 
-    res = _parsedate(datetimestring, fuzzy=fuzzy,
-        dayfirst=dayfirst, yearfirst=yearfirst, locale=locale)
+    res = _parsedate(datetimestring; args...)
 
     # Fill in default values if none exits
-    res["year"] = convertyear(get(res, "year", year(default)))
-    get!(res, "month", month(default))
-    get!(res, "day", day(default))
+    get!(res, "year", Year(default))
+    get!(res, "month", Month(default))
+    get!(res, "day", Day(default))
 
-    return Date(res["year"], res["month"], res["day"])
+    delete!(res, "timezone")
+    return Date(values(res)...)
 end
 
 function _parsedate(datetimestring::AbstractString; fuzzy::Bool=false,
-    timezone_infos::Dict{AbstractString, TimeZone}=Dict{AbstractString, TimeZone}(),
-    dayfirst::Bool=false,
-    yearfirst::Bool=false,
-    locale::AbstractString="english",
+    timezone_infos::Dict{AbstractString, TimeZone}=Dict{AbstractString, TimeZone}(), # Specify what a timezone is
+    dayfirst::Bool=false, # MM-DD-YY vs DD-MM-YY
+    yearfirst::Bool=false, # MM-DD-YY vs YY-MM-DD
+    locale::AbstractString="english", # Locale in Dates.VALUETOMONTH and VALUETODAYOFWEEK
 )
-    month = monthtovalue(locale)
-    weekday = weekdaytovalue(locale)
+    res = Dict{AbstractString, Union{Period,TimeZone}}()
```

Comment:
Seems like you may just want to make `res` a custom type. Something like:

```
type Parts
    year::Nullable{Year}
    ....
end

Base.convert{T}(::Type{Nullable{T}}, x::Any) = Nullable{T}(T(x))
```

Then you could do things like:

```
res.year = 1
```